src/middleware.rs

```diff
@@ -0,0 +1,53 @@
+use axum::{
+    body::Body,
+    extract::Request,
+    http::StatusCode,
+    middleware::Next,
+    response::{IntoResponse, Response},
+};
+use core::str;
+use http_body_util::BodyExt;
+use log::{debug, log_enabled, Level};
+
+pub async fn body_logger_middleware(
+    request: Request,
+    next: Next,
+) -> Result<impl IntoResponse, Response> {
+    let request = buffer_request_body(request).await?;
+    let response = next.run(request).await;
+    let response = buffer_response_body(response).await?;
+
+    Ok(response)
+}
+
+async fn buffer_request_body(request: Request) -> Result<Request, Response> {
+    let (parts, body) = request.into_parts();
+
+    let bytes = body
+        .collect()
+        .await
+        .map_err(|err| (StatusCode::INTERNAL_SERVER_ERROR, err.to_string()).into_response())?
+        .to_bytes();
+
+    if log_enabled!(target: "live_compositor::routes", Level::Debug) {
```

Comment:
there is already file route that use sth that does not already exist e.g. `live_compositor::log_request_body`